### ISSUE TITLE
TypeError: Cannot read property 'some' of undefined 

### ISSUE THREAD

**Issue by felinto-dev** [nestjs/graphql#1387]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

I'm trying to use "GraphQL CLI Plugin" in e2e tests following this tutorial:
https://docs.nestjs.com/graphql/cli-plugin#integration-with-ts-jest-e2e-tests
But the code does not work because when I run the e2e tests I get this error:

```js
api_1  | > jest --config ./test/jest-e2e.json
api_1  |
api_1  | FAIL test/modules/auth/auth.e2e-spec.ts
api_1  |   ● Test suite failed to run
api_1  |
api_1  |     TypeError: Cannot read property 'some' of undefined
api_1  |
api_1  |       4 | module.exports.name = 'nestjs-graphql-transformer';
api_1  |       5 | module.exports.version = 1;
api_1  |     > 6 | module.exports.factory = (cs) => transformer.before({}, cs.tsCompiler.program);
api_1  |         |                                                            ^
api_1  |       7 |
api_1  |
api_1  |       at ConfigSet.isTestFile (../node_modules/ts-jest/dist/config/config-set.js:436:40)
api_1  |       at ../node_modules/ts-jest/dist/compiler/language-service.js:116:55
api_1  |           at Array.filter (<anonymous>)
api_1  |       at Object.exports.initializeLanguageServiceInstance (../node_modules/ts-jest/dist/compiler/language-service.js:116:10)
api_1  |       at Object.exports.createCompilerInstance (../node_modules/ts-jest/dist/compiler/instance.js:51:30)
api_1  |       at ConfigSet.get (../node_modules/ts-jest/dist/config/config-set.js:396:31)
api_1  |       at ConfigSet.<anonymous> (../node_modules/ts-jest/dist/utils/memoize.js:44:24)
api_1  |       at Object.module.exports.factory (test/ast.ts:6:60)
api_1  |       at ../node_modules/ts-jest/dist/config/config-set.js:267:61
api_1  |           at Array.map (<anonymous>)
```

These are some relevant files about this issue:

**jest-e2e.json**:
```json
{
  "moduleDirectories": [
    "node_modules",
    "src"
  ],
  "moduleFileExtensions": ["js", "json", "ts"],
  "modulePathIgnorePatterns": ["<rootDir>/dist"],
  "moduleNameMapper": {
    "@/(.*)": "<rootDir>/src/$1"
  },
  "rootDir": "..",
  "testEnvironment": "node",
  "testRegex": ".e2e-spec.ts$",
  "transform": {
    "^.+\\.(t|j)s$": "ts-jest"
  },
  "globals": {
    "ts-jest": {
      "astTransformers": {
        "before": ["<rootDir>/test/ast.ts"]
      }
    }
  }
}
```

**ast.ts**:
```ts
// eslint-disable-next-line @typescript-eslint/no-var-requires
const transformer = require('@nestjs/graphql/plugin');

module.exports.name = 'nestjs-graphql-transformer';
module.exports.version = 1;
module.exports.factory = (cs) => transformer.before({}, cs.tsCompiler.program);
```

**package.json**:
```json
    "test:e2e": "npm run build && jest --config ./test/jest-e2e.json",
```

## Environment

<pre><code>
Nest version: 7.5
 
For Tooling issues:
- Node version: 15.15.5
- Platform:  Windows

Others:
@nestjs/graphq: 7.9.8
ts-jest: 26.5.0
ts-node: 9.1.1
</code></pre>

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by felinto-dev** [comment]
> Please provide a minimum reproduction repository.

You need to have Docker installed to replicate this issue.
1. Unzip the package
2. Change the current directory to the zip directory
3. npm i
4. npm run test:e2e-docker

[repo-fntwork.zip](https://github.com/nestjs/graphql/files/5989290/repo-fntwork.zip)

**Comment by kamilmysliwiec** [comment]
Could you please create a Github repository (instead of sending a .zip file)?

**Comment by felinto-dev** [comment]
It was not my intention for this project to be public, but ok, I did it
https://github.com/felinto-dev/repo-fntwork-nestjs-issue

**Comment by felinto-dev** [comment]
I am able to replicate the bug

I used this nests sample project as minimal repository.
https://github.com/nestjs/nest/tree/master/sample/23-graphql-code-first

Please, check the git commit to check what files I changed to reproduce this issue.
https://github.com/felinto-dev/tscompiler-nestjs-issue

Please, let me know if you are able to replicate this issue.

**Comment by CatsMiaow** [comment]
I confirmed it as a problem with the `ts-jest` version.

`v26.4.4` no problem
`v26.5.1` error

This is a different story, but with `nestjs-graphql-transformer` the test speed is very slow. Perhaps cache issue?

**Comment by kamilmysliwiec** [comment]
Can we report this issue in the `ts-jest` repository then? It seems that adding a check whether `this._matchablePatterns` property is not `undefined` in the `ConfigSet#isTestFile` would fix this issue (`config-set.js` file).

**Comment by CatsMiaow** [comment]
Created an issue. https://github.com/kulshekhar/ts-jest/issues/2371
There seems to be a code change in ts-jest `v26.5.x`.

**Comment by kamilmysliwiec** [comment]
Thanks! Let's track this here https://github.com/kulshekhar/ts-jest/issues/2371

**Comment by CatsMiaow** [comment]
Fixed in ts-jest `v26.5.2`
